persistent-sqlite/Database/Persist/Sqlite.hs

```diff
@@ -64,7 +64,12 @@ open' connStr logFunc = do
             | Just cs <- T.stripPrefix "WAL=off " connStr -> (cs, False)
             | otherwise                                   -> (connStr, True)
 
-    conn <- Sqlite.open connStr'
+    let (connStr'', readOnly) = case () of
+          ()
+            | Just cs <- T.stripPrefix "READONLY " connStr'  -> (cs, True)
+            | Just cs <- T.stripPrefix "READWRITE " connStr' -> (cs, False)
+            | otherwise                                      -> (connStr', False)
+    conn <- Sqlite.open connStr'' readOnly
```

Comment:
Alternatively, would it be sufficient to just check for a substring to lookup the config option, and remove it if present? That wouldn't be sensitive to argument order.